### PR TITLE
 support ttl db

### PR DESCRIPTION
**Pull request by zhangsoledad**
### What is changed?

* Add ttl db support 
* fix Support UTF-8 file paths on Windows (fetch from upstream)
* add debug impl for db_pinnable_slice

### Check List 

Tests 

- Unit test
https://github.com/nervosnetwork/rust-rocksdb/blob/a4db365c51a83f729256f4306864ce065cf841f9/tests/test_ttl_db.rs

### CODE REVIEW

File: librocksdb-sys/build.rs [hunk 1]
```diff
@@ -111,41 +129,46 @@ fn build_rocksdb() {
     }
 
     if target.contains("darwin") {
-        config.define("OS_MACOSX", Some("1"));
-        config.define("ROCKSDB_PLATFORM_POSIX", Some("1"));
-        config.define("ROCKSDB_LIB_IO_POSIX", Some("1"));
+        config.define("OS_MACOSX", None);
```
Comment (doitian) [inline librocksdb-sys/build.rs:126]: None means `-DOS_MACOSX` ?

Comment (zhangsoledad) [reply librocksdb-sys/build.rs:126]: Yes，pass None as the value for a value-less define.
https://rust-lang-nursery.github.io/rust-cookbook/development_tools/build_tools.html#compile-a-c-library-while-setting-custom-defines.

Comment (zhangsoledad) [reply librocksdb-sys/build.rs:126]: This fetch from upstream.

**Review by chaoticlonghair — APPROVED** [review]

File: librocksdb-sys/build.rs [hunk 2]
```diff
@@ -308,7 +340,19 @@ fn try_to_find_and_link_lib(lib_name: &str) -> bool {
...
+        if !cxx_std.starts_with("-std=") {
+            format!("-std={}", cxx_std)
+        } else {
+            cxx_std
+        }
+    })
+}
+
 fn main() {
+    bindgen_rocksdb();
```
Comment (chaoticlonghair) [inline librocksdb-sys/build.rs:354]: Add one more function to bindgen for [`librocksdb-sys/src/patches.rs`](https://github.com/nervosnetwork/rust-rocksdb/blob/f42d8ac/librocksdb-sys/src/patches.rss) too.

**Review by chaoticlonghair — CHANGES_REQUESTED** [review]

File: librocksdb-sys/build.rs (hunk 2 above)
Comment (zhangsoledad) [reply librocksdb-sys/build.rs:354]: All bundle in `bindings.rs`. https://github.com/nervosnetwork/rust-rocksdb/compare/f42d8acd4680a74acd5af998cabc484db653a8de..1fdc1e1c310e2ca578213ef3df802cef2b4b2a3e

**Review by chaoticlonghair — APPROVED** [review]

**Pull request merged**